insights/client/__init__.py

```diff
@@ -523,4 +523,10 @@ def collect():
 
 def upload():
     egg_path = sys.stdin.read().strip()
-    run("upload", egg_path)
+    if not config["to_json"]:
+        logger.debug("Flag --to-json not used. Running upload.")
+        run("upload", egg_path)
+    else:
+        logger.debug("Flag --to-json used. Printing upload results.")
+        print "INIT: %s" % (str(run("upload", egg_path)))
+        return
```

Comment:
do we need to return at the end of the function?

Reply:
Maybe? I was just replicating other instances where we simply 'return' back to the wrapper RPM.